app/API/V1/Repositories/LoginAttemptRepository.php

```diff
@@ -0,0 +1,228 @@
+    {
+        $current = Carbon::now();
+        $max_partial_lock = (int) env('MAX_LOGIN_ATTEMPTS_BEFORE_PARTIAL_LOCK', 0);
+        $max_full_lock = (int) env('MAX_LOGIN_ATTEMPTS_BEFORE_FULL_LOCK', 0);
+        $partialLockTimeOut = (int) env('LOGIN_PARTIAL_LOCK_TIMEOUT', 0);
+        $fullLockTimeOut = (int) env('LOGIN_FULL_LOCK_TIMEOUT', 0);
+        $diffSec = $current->diffInSeconds($attempt->getUpdatedAt());
+
+        /** Check for attempts */
+        if ($max_partial_lock && $attempt->getCount() >= $max_partial_lock) {
+            if ($partialLockTimeOut && $diffSec >= $partialLockTimeOut) {
+                $attempt = $this->resetAttempt($attempt, $attempt->getFullLockCount() + 1);
+                if ($max_full_lock && $attempt->getFullLockCount() >= $max_full_lock) {
+                    return self::LOGIN_ATTEMPT_ERROR_ACCOUNT_FULL_LOCKED;
+                } else {
+                    return self::LOGIN_ATTEMPT_DEFAULT;
+                }
+            } else {
+                if ($attempt->getCount() >= $max_partial_lock) {
+                    $attempt = $this->resetAttempt($attempt, $attempt->getFullLockCount() + 1);
+                    if ($max_full_lock && $attempt->getFullLockCount() >= $max_full_lock) {
+                        return self::LOGIN_ATTEMPT_ERROR_ACCOUNT_FULL_LOCKED;
+                    } else {
+                        return self::LOGIN_ATTEMPT_ERROR_ACCOUNT_PARTIAL_LOCKED;
+                    }
+                } else {
+                    return self::LOGIN_ATTEMPT_ERROR_ACCOUNT_PARTIAL_LOCKED;
+                }
+            }
+        } else if ($max_full_lock && $attempt->getFullLockCount() >= $max_full_lock) {
+            if ($fullLockTimeOut && $diffSec >= $fullLockTimeOut) {
+                $this->resetAttempt($attempt);
+                return self::LOGIN_ATTEMPT_DEFAULT;
+            } else {
+                $this->lockUser($attempt->getUser(), true);
+                return self::LOGIN_ATTEMPT_ERROR_ACCOUNT_FULL_LOCKED;
+            }
+        } else {
+            return self::LOGIN_ATTEMPT_DEFAULT;
+        }
+    }
+
+    /**
+     * Attempt Check
+     *
+     * @param User $user
+     * @return int
+     * @throws \Doctrine\DBAL\ConnectionException
+     * @throws \Doctrine\ORM\OptimisticLockException
+     */
+    public function attemptCheck(User $user)
+    {
+        /** @var LoginAttempt $attempt */
+        $attempt = $this->findOneBy(['user'=>$user]);
+        return $attempt ? $this->getAttemptStatus($attempt) : self::LOGIN_ATTEMPT_DEFAULT;
+    }
+
+
+    /**
+     * @return array
+     */
+    public function getTTConfig(): array
+    {
+        return [
+            'default'=>[
+                'read'=>[
+                    'permissions'=>[
+                        'allowed'=>true
+                    ]
+                ]
+            ],
+            'superAdmin'=>[
+                'extends'=>[':default'],
+                'read'=>[
+                    'permissions'=>[
```

Comment:
Since this extends default, you don't need to put the permissions in again.

Reply:
@jerazo though is the extends at the wrong level here? I think the extends would go in side the read block, not in the superAdmin block

Reply:
Fixed